src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -78,12 +80,158 @@ public override unsafe int GetByteCount(char* chars, int count)
             return EncodingForwarder.GetByteCount(this, chars, count);
         }
 
-        public override int GetBytes(String chars, int charIndex, int charCount,
-                                              byte[] bytes, int byteIndex)
+        public unsafe override byte[] GetBytes(String s)
+        {
+            if (s == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.s, ExceptionResource.ArgumentNull_String);
+            Contract.EndContractBlock();
+
+            int charCount = s.Length;
+
+            byte[] bytes;
+            if (charCount > 0) 
+            {
+                fixed (char* input = s)
+                    bytes = GetBytesValidated(input, charCount);
+            } 
+            else 
+            {
+                bytes = Array.Empty<byte>();
+            }
+
+            return bytes;
+        }
+
+        public unsafe override int GetBytes(String chars, int charIndex, int charCount, byte[] bytes, int byteIndex)
+        {
+            if ((chars == null) ||
+                (bytes == null) ||
+                (charIndex < 0) ||
+                (charCount < 0) ||
+                (chars.Length - charIndex < charCount) ||
+                (byteIndex < 0 || byteIndex > bytes.Length))
```

Comment:
runs if `charCount == 0 && byteCount == 0` and throws a different error if `charCount > 0 && byteCount == 0` I could probably roll it into the same Throw call...